plugin-server/src/main/pluginsServer.ts

```diff
@@ -101,6 +101,16 @@ export async function startPluginsServer(
         shuttingDown = true
         status.info('💤', ' Shutting down gracefully...')
         lastActivityCheck && clearInterval(lastActivityCheck)
+
+        // HACKY: Stop all consumers and the graphile worker, as well as the
+        // http server. Note that we close the http server before the others to
+        // ensure that e.g. if something goes wrong and we deadlock, then if
+        // we're running in k8s, the liveness check will fail, and thus k8s will
+        // kill the pod.
+        //
+        // I say hacky because we've got a weak dependency on the liveness check
+        // configuration.
+        httpServer?.close()
```

Comment:
This makes sense to me. 
![image](https://user-images.githubusercontent.com/391319/200612576-233389f0-701e-40fc-9b33-7c0e7ab06bac.png)